frontend/src/v5/ui/controls/popover/popover.component.tsx

```diff
@@ -0,0 +1,42 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ReactElement, useState } from 'react';
+import { PopoverContainer } from './popover.styles';
+
+type IPopover = {
+	anchor: (props) => ReactElement;
+	popoverContent: () => ReactElement;
+};
+
+export const Popover = ({ anchor: AnchorEl, popoverContent: PopoverContent }: IPopover) => {
+	const [anchorEl, setAnchorEl] = useState<Element | null>(null);
+	const onMouseEnter = (event) => setAnchorEl(event.currentTarget);
+	const onMouseLeave = () => setAnchorEl(null);
+
+	return (
+		<>
```

Comment:
This is just an idea, but you could re-write the component as follows (without the inline styling):
```ts
<div style={{ display: 'inline-block' }} onMouseEnter={onMouseEnter} onMouseLeave={onMouseLeave}>
	<AnchorEl />
	<PopoverContainer
		open={!!anchorEl}
		anchorEl={anchorEl}
	>
		{children}
	</PopoverContainer>
</div>
```
This way you do not risk to override the onMouseEnter and onMouseLeave properties of an anchorEl